src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/util/TokenTestUtils.java

```diff
@@ -68,24 +61,63 @@ public static TokenInfo createTestTokenInfo(String friendlyName) {
      * - tokenStatus = OK
      * - keyInfos = empty
      * - tokenInfo map = empty
-     * @param friendlyName
-     * @param id
-     * @return
      */
-    public static TokenInfo createTestTokenInfo(String friendlyName, String id) {
-        TokenInfo tokenInfo = new TokenInfo(TokenInfo.SOFTWARE_MODULE_TYPE,
-                friendlyName,
-                id,
-                false,
-                true,
-                true,
-                "serial-number",
-                "label",
-                123,
-                TokenStatusInfo.OK,
-                new ArrayList<>(),
-                new HashMap<>());
-        return tokenInfo;
+    public static class TokenInfoBuilder {
```

Comment:
Maybe Lombok's [`@Builder` annotation](https://projectlombok.org/features/Builder) could have been utilized here? I am not 100% sure if it fits this case but if it does it removes a lot of lines from this inner class. Same with [CertificateInfoBuilder](https://github.com/nordic-institute/X-Road/pull/294/files#diff-6a455e99ba4ac4bc2b4a0c614062ad6cR293). Just a thought though

Reply:
Good idea, I will try if it works here. Always good to remove boilerplate if it's possible. Although Lombok has also it's downsides - but this is tests where we don't need to stress so much about those.

Reply:
I tried this a bit, and I think I am rolling back the attempt. It was close but these details were inconvenient:
- I would rather not change ee.ria.xroad.signer.protocol.dto objects like KeyInfo since a) builder is a test-only concern at least right now, and b) it looks like there's some complexity involved with combining e.g. both @Value and @Builder annotations in the same class https://projectlombok.org/features/Builder - basically, I am hesitant to change those dto objects since that would mean a lot of testing to ensure nothing was broken in other modules
- If dto objects themselves cannot be annotated with @Builder, you would have to annotate a method on TokenTestUtils, and you cannot achieve the same kind of default-value handling as current implementation has that way as far as I can see 

Basically it started to look like too much work for something that would be a positive change, but is not really not broken at the moment. It looked nice though so I will keep that in mind when doing the next builder

```
    @Builder
    public static KeyInfo newKeyInfo(String id,
            String friendlyName,
            KeyUsageInfo keyUsageInfo,
            @Singular List<CertRequestInfo> certRequests) {
            return new KeyInfo(true,
                    keyUsageInfo,
                    friendlyName,
                    id,
                    "label",
                    "public-key",
                    new ArrayList(),
                    certRequests,
                    "sign-mechanism-name");
    }
```